internal/local/app_test.go

```diff
@@ -111,46 +134,48 @@ func TestFindApp(t *testing.T) {
 			})
 
 			for _, tcInner := range []struct {
-				name    string
-				appData AppData
+				description string
+				name        string
+				appData     AppData
 			}{
-				{"local", config.appDataLocal},
-				{"remote", config.remoteAppData},
+				{"and a working directory at the root of a local project", "local", config.appDataLocal},
+				{"and a working directory at the root of a remote project", "remote", config.remoteAppData},
+				{"and a nested working directory containing another config", "nested/graphql", config.nestedAppData},
 			} {
-				t.Run(fmt.Sprintf("and a working directory at the root of a %s project", tcInner.name), func(t *testing.T) {
-					projectRoot := filepath.Join(testRoot, tcInner.name)
+				t.Run(tcInner.description, func(t *testing.T) {
+					path := filepath.Join(testRoot, tcInner.name)
 
-					_, insideProject, err := FindApp(projectRoot)
+					_, insideProject, err := FindApp(path)
 					assert.Nil(t, err)
 					assert.True(t, insideProject, "should be inside project")
 
-					app, err := LoadApp(projectRoot)
+					app, err := LoadApp(path)
 					assert.Nil(t, err)
 					assert.Equal(t, tcInner.appData, app.AppData)
+					assert.Equal(t, tcInner.appData.ConfigVersion(), app.ConfigVersion())
 				})
 			}
 
-			t.Run("And a working directory at the project root return the correct data", func(t *testing.T) {
+			t.Run("and an empty project should return an error when finding app", func(t *testing.T) {
 				projectRoot := filepath.Join(testRoot, "empty")
 
 				_, insideProject, err := FindApp(projectRoot)
-				assert.Nil(t, err)
-				assert.True(t, insideProject, "should be inside project")
+				assert.Equal(t, errFailedToParseAppConfig(filepath.Join(projectRoot, config.file.String())), err)
+				assert.False(t, insideProject, "should not be found")
 
 				app, err := LoadApp(projectRoot)
 				assert.Equal(t, errFailedToParseAppConfig(filepath.Join(projectRoot, config.file.String())), err)
 				assert.Equal(t, App{}, app)
 			})
 
-			t.Run("And an empty project should return an error when loading app", func(t *testing.T) {
-				projectRoot := filepath.Join(testRoot, "empty")
-
-				_, insideProject, err := FindApp(projectRoot)
+			t.Run("and an invalid project should fail to find app", func(t *testing.T) {
+				path := filepath.Join(testRoot, config.invalidAppData.Name())
+				_, foundApp, err := FindApp(path)
 				assert.Nil(t, err)
-				assert.True(t, insideProject, "should be inside project")
+				assert.False(t, foundApp, "should not be found")
 
-				app, err := LoadApp(projectRoot)
-				assert.Equal(t, errFailedToParseAppConfig(filepath.Join(projectRoot, config.file.String())), err)
+				app, err := LoadApp(path)
+				assert.Nil(t, err)
```

Comment:
[question] what are you thoughts on this being an error scenario instead?  it seems fairly similar to the "empty" case, I'm trying to wrap my head around why they behave differently here

Reply:
difference between the invalid case and the empty case:
--> the empty case will encounter an error when `LoadConfig` tries to read/parse the file [here](https://github.com/10gen/realm-cli/blob/29820cfb35f206efdf45983381e4acf607d686d8/internal/local/app.go#L230).
--> the invalid case will have its config file parsed without error but will not be recognized as an app path since it doesn't contain a real config 
```
{
  "garbage": 1
}
```

Reply:
i may just be tripped up on names here... correct me if i'm wrong here, but the "empty case" fails b/c we're trying to parse invalid json syntax, right?  and the "invalid case" doesn't error but fails to find app b/c the JSON that was parsed simply didn't have "config_version" as a field?

if that's so, maybe we can just tweak the naming in the test here: where "empty" is actually "invalid json" and "invalid" is actually "missing config version"?  curious as to your thoughts on that one

Reply:
agreed! I'll change it